### PR TITLE
hide get_cpu_infos

### PR DESCRIPTION
**Pull request by Christian-B**
Part of https://github.com/SpiNNakerManchester/SpiNNMan/pull/358

### CODE REVIEW

File: spinn_front_end_common/interface/abstract_spinnaker_base.py [hunk 1]
```diff
@@ -2270,8 +2270,8 @@ def __recover_from_error(self, exception):
         if not unsuccessful_cores:
             for executable_type, core_subsets in \
                     self._data_writer.get_executable_types().items():
-                failed_cores = transceiver.get_cpu_infos(
-                    core_subsets, executable_type.end_state, False)
+                failed_cores = transceiver.get_cores_in_states(
```
Comment (rowleya) [inline spinn_front_end_common/interface/abstract_spinnaker_base.py:2273]: Should this be "not in state" since it was include=False?

File: spinn_front_end_common/interface/interface_functions/application_finisher.py [hunk 2]
```diff
@@ -62,16 +62,16 @@ def application_finisher():
...
 
         txrx.send_signal(app_id, Signal.SYNC0)
         txrx.send_signal(app_id, Signal.SYNC1)
 
         for core_subset in all_core_subsets:
             for processor in core_subset.processor_ids:
-                if not successful_cores_finished.is_core(
-                        core_subset.x, core_subset.y, processor):
+                if (core_subset.x, core_subset.y, processor) not in \
+                        successful_cores_finished:
```
Comment (rowleya) [inline spinn_front_end_common/interface/interface_functions/application_finisher.py:74]: Is that change necessary?  It seems that the function get_cores_in_states returns the same information as get_cpu_infos, but it has now been put in a set instead of using the built-in set...

File: spinn_front_end_common/interface/interface_functions/chip_provenance_updater.py [hunk 3]
```diff
@@ -59,15 +59,17 @@ def _run(self):
...
-        if error_cores or watchdog_cores or idle_cores:
+            [CPUState.RUN_TIME_EXCEPTION, CPUState.WATCHDOG, CPUState.IDLE])
+
+        if bad_cores:
+            error_cores = self.__txrx.get_cores_in_states(
+                self.__all_cores, CPUState.RUN_TIME_EXCEPTION)
+            watchdog_cores = self.__txrx.get_cores_in_states(
+                self.__all_cores, CPUState.WATCHDOG)
+            idle_cores = self.__txrx.get_cores_in_states(
+                self.__all_cores, CPUState.IDLE)
```
Comment (rowleya) [inline spinn_front_end_common/interface/interface_functions/chip_provenance_updater.py:72]: Similarly to a previous comment, get_cores_in_states returns the same information as get_cpu_infos so can't you still do the same thing rather than get them all again?

File: spinn_front_end_common/utilities/emergency_recovery.py [hunk 4]
```diff
@@ -34,36 +34,23 @@ def _emergency_state_check():
...
         if rte_count or watchdog_count:
-            states = txrx.get_cpu_infos(
-                None, [CPUState.RUN_TIME_EXCEPTION, CPUState.WATCHDOG], True)
+            bad_cores = txrx.get_cores_in_states(
+                None, [CPUState.RUN_TIME_EXCEPTION, CPUState.WATCHDOG])
             logger.warning(
                 "unexpected core states (rte={}, wdog={})",
                 rte_count, watchdog_count)
-            logger.warning(states.get_status_string())
+            logger.warning("cores {}", set(bad_cores))
```
Comment (rowleya) [inline spinn_front_end_common/utilities/emergency_recovery.py:42]: I think this can still get the status string?

File: spinn_front_end_common/utilities/emergency_recovery.py [hunk 5]
```diff
@@ -34,36 +34,23 @@ def _emergency_state_check():
...
-                        chip.x, chip.y, p)
-                    if info.state in (
-                            CPUState.RUN_TIME_EXCEPTION, CPUState.WATCHDOG):
-                        infos.add_processor(chip.x, chip.y, p, info)
-                except Exception:
-                    errors.append((chip.x, chip.y, p))
-        if len(infos):
-            logger.warning(infos.get_status_string())
+        errors = txrx.get_cores_in_states(
+            CPUState.RUN_TIME_EXCEPTION, CPUState.WATCHDOG)
```
Comment (rowleya) [inline spinn_front_end_common/utilities/emergency_recovery.py:47]: This is a little more subtle - I think the idea was in the case of an error to go chip-by-chip to avoid getting a load of errors all at once

Comment (rowleya) [reply spinn_front_end_common/utilities/emergency_recovery.py:47]: Also note errors is when things go wrong in reading the state

File: spinn_front_end_common/utilities/emergency_recovery.py [hunk 6]
```diff
@@ -34,36 +34,23 @@ def _emergency_state_check():
...
         if len(errors) > 10:
             logger.warning(
                 "Could not read information from {} cores", len(errors))
         else:
             logger.warning(
-                "Could not read information from cores {}", errors)
+                "Could not read information from cores {}", set(errors))
```
Comment (rowleya) [inline spinn_front_end_common/utilities/emergency_recovery.py:53]: See above about the change in meaning of errors

File: spinn_front_end_common/utilities/system_control_logic.py [hunk 7]
```diff
@@ -163,13 +163,12 @@ def _load_application(executable_targets, app_id):
     # Check that the binaries have reached a wait state
     count = transceiver.get_core_state_count(app_id, CPUState.READY)
     if count < executable_targets.total_processors:
-        cores_ready = transceiver.get_cpu_infos(
-            executable_targets.all_core_subsets, [CPUState.READY], False)
+        cores_ready = transceiver.get_cores_in_states(
+            executable_targets.all_core_subsets, [CPUState.READY])
```
Comment (rowleya) [inline spinn_front_end_common/utilities/system_control_logic.py:167]: Interesting - in master this seems to hold "cores_ready" but actually it is "cores_not_ready"...  I think it is meant to show the status of things that are not ready though... (and then it should be able to use the status string)

File: spinn_front_end_common/utility_models/data_speed_up_packet_gatherer_machine_vertex.py [hunk 8]
```diff
@@ -1000,10 +1000,11 @@ def unset_cores_for_data_streaming(self):
                 FecDataView.iterate_monitors())
             try:
                 transceiver = FecDataView.get_transceiver()
-                error_cores = transceiver.get_cpu_infos(
-                    core_subsets, {CPUState.RUNNING}, False)
+                error_cores = transceiver.get_cores_in_states(
+                    core_subsets, {CPUState.RUNNING})
```
Comment (rowleya) [inline spinn_front_end_common/utility_models/data_speed_up_packet_gatherer_machine_vertex.py:1004]: Should be not in state

**Review by rowleya — CHANGES_REQUESTED** [review]
Quite a lot of these changes are hard to test because they only happen on errors which then haven't happened in testing!  I believe that get_cores_in_states and get_cores_not_in_states still return the same thing as get_cpu_infos, so it shouldn't be necessary to wrap them in a set...

File: spinn_front_end_common/interface/abstract_spinnaker_base.py (hunk 1 above)
Comment (Christian-B) [reply spinn_front_end_common/interface/abstract_spinnaker_base.py:2273]: agreed

File: spinn_front_end_common/interface/interface_functions/application_finisher.py (hunk 2 above)
Comment (Christian-B) [reply spinn_front_end_common/interface/interface_functions/application_finisher.py:74]: The difference is that get_cpu_infos promises to return a spinnman.model.CPUInfos while get_cores_(not_)in_states only contracts to return an iterable(int, int, int) xyp.   This gives Spinn2 the freedom the have a different "CPUInfos" model

File: spinn_front_end_common/utilities/emergency_recovery.py (hunk 5 above)
Comment (Christian-B) [reply spinn_front_end_common/utilities/emergency_recovery.py:47]: Yes the idea was to go chip by chip. BUT txrx.get_cpu_information_from_core did not support that.
Instead it gets all states and then returns the one wanted.

**Comment by Christian-B** [comment]
Replaced with https://github.com/SpiNNakerManchester/SpiNNMan/pull/366
